### ISSUE TITLE
described files / directions do not exist

### ISSUE THREAD

**Issue by Mullerinho** [lmartinking/webscreensaver#34]
Hey there, have a few challenges:

1) /usr/lib/xscreensaver does not exist

I found the /usr/libexec/xscreensaver directory and copied the webscreensaver file inside.

2) In the direction /usr/libexec/xscreensaver is no .xscreensaver file. I found it at the home directory, added the webscreensaver                  \n\
However, I can't choose webscreensaver at the xscreensaver GUI (also after running the python file)

3) Where exactly should I add the URL I wan't
    -url <url_to_the_page_you_want>?

It would be very well, if you could help me, because a webpage as a screensaver is amazing! Thank you in advance

**Comment by lmartinking** [comment]
Hi @Mullerinho it depends on the distribution. The location `/usr/libexec/xscreensaver` is more "correct" as it contains executables rather than libraries.

As for `.webscreensaver`, if you look in the Readme it is proceeded by a `~/` which signifies your home directory. However I'll update the readme for clarity.

Unfortunately editing a screensaver settings using the XScreensaver GUI requires an extra XML file which I have not bothered to write. I took the simple approach and gave instructions for the command line options.

You need to edit your XScreensaver config file. It has a particular format.

**Comment by Mullerinho** [comment]
Hey Imartinking, thank you very much for the fast response! Yet I am one step further. I can now choose the webscreensaver. Unfortunately this is the output (see picture). Can you help me?

With XScreensaver config file you mean the .xscreensaver file?

Where do I have to insert the URL, I want to?


![webscreensaver](https://github.com/lmartinking/webscreensaver/assets/160370952/877f6078-2815-4044-a45a-24e1f6e04000)

**Comment by lmartinking** [comment]
The `.xscreensaver` file will be located in your home directory. Eg: `/home/mullerinho/.xscreensaver`.

**Comment by Mullerinho** [comment]
Ok, I tooked the file and changed it to this (see picture). Is this right? F.e. I tooked www.google.de

I still got the error messages, I send on the first picture. What does it mean? PyGTKDeprecationWarning + three WebKit2.Settings messages?

PS.: Yes I installed     sudo apt install python3 python3-gi gir1.2-webkit2-4.0 gir1.2-gtk-3.0

![Screenshot from 2024-03-04 12-02-06](https://github.com/lmartinking/webscreensaver/assets/160370952/22107848-4d78-41bc-bb4b-7d7995852b65)

**Comment by lmartinking** [comment]
@Mullerinho which distribution are you using? I will see if I can reproduce the issue.

**Comment by Mullerinho** [comment]
Thank you for your support! :)

![Screenshot from 2024-03-05 11-02-27](https://github.com/lmartinking/webscreensaver/assets/160370952/caacc5c4-4075-40c0-8b55-0fd39d480a33)

**Comment by lmartinking** [comment]
@Mullerinho I've spent an hour trying to debug the issue, and I think it's related to how Gnome handles windows. Gnome does use its own compositor, which I can't disable.

The stand-alone window works, however running `webscreensaver` via the XScreensaver prefs preview or via XScreensaver itself fails to work correctly (blank screen).

Unfortunately, I can't investigate this further as there is a huge stack to dig into: Gnome, as well as GTK and GTKWebKit.

**Comment by Mullerinho** [comment]
Ok, thats annoying! However, thank you so much for trying to help me!!! :-)

**Comment by lmartinking** [comment]
@Mullerinho no worries! thanks for all your logs, etc. You could consider trying XFCE.

**Comment by Mullerinho** [comment]
Yes, I was also thinking about it. I'll try in it in the next time and maybe update you :)
